### PR TITLE
a Simple and Fun Quiz App

### PR DESCRIPTION
**Pull request by Angryman18**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Angryman18** [comment]
Its really a dirty work and dirty code also.

**Review by Programming-School-Pro-Coding — DISMISSED** [review]
Very Amazing

**Comment by atapas** [comment]
This is one of the most incredible play I've seen! I shall start doing code reviews soon.

**Comment by Programming-School-Pro-Coding** [comment]
> This is one of the most incredible play I've seen! I shall start doing code reviews soon.

Give me the reviewing i am free

**Comment by Angryman18** [comment]
> This is one of the most incredible play I've seen! I shall start doing code reviews soon.

Thanks. I see a bug. Will fix at next push.

**Comment by Angryman18** [comment]
sorry i was mistaken about the bug. i think its working good. although i fixed a "key" was missing while mapping array.

**Comment by atapas** [comment]
@Angryman18 Could you please create an issue for the play and associate with the PR to start the review?

**Comment by Angryman18** [comment]
> @Angryman18 Could you please create an issue for the play and associate with the PR to start the review?

Created sir.

**Comment by atapas** [comment]
> > @Angryman18 Could you please create an issue for the play and associate with the PR to start the review?
> 
> Created sir.

Thanks but it was not done correctly.

So next time,

- You need to create a play request if it is a play, else a bug or feature.
- After creating the issue you can attach it to a PR(if already exist) from the right side setting of an issue.

**Comment by Angryman18** [comment]
ah. sorry about that. not very good on github yet but i learned a lot in recent days.

**Comment by atapas** [comment]
> > This is one of the most incredible play I've seen! I shall start doing code reviews soon.
> 
> Give me the reviewing and merging i am free

Please go ahead and finish the review

**Review by Programming-School-Pro-Coding — DISMISSED** [review]
LGTM

**Comment by atapas** [comment]
> LGTM

Thanks, @koustov please finish your review too.

File: src/plays/fun-quiz/FrontScreen.scss [hunk 1]
```diff
@@ -0,0 +1,174 @@
+
+@import url('https://fonts.googleapis.com/css2?family=Quicksand:wght@300;400;500;600;700&display=swap');
```
Comment (atapas) [inline src/plays/fun-quiz/FrontScreen.scss:2]: Do you need this font? @Angryman18 

If yes, Can you please make make sure it doesn't override the font of reactplay when switch between this play and others?

Comment (Angryman18) [reply src/plays/fun-quiz/FrontScreen.scss]: Yes its removed.

**Comment by koustov** [comment]
Started reviewing. I will finish it soon.

**Review by koustov — CHANGES_REQUESTED** [review]

**Comment by koustov** [comment]
@Angryman18 this is a great play ❤. I am fully energized to see the granularity maintained here.

**Comment by Angryman18** [comment]
ok i will fix by night

File: src/plays/fun-quiz/EndScreen.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+const EndScreen = ({ quizSummary, redirectHome }) => {
+  const { correctAnswers, cheatedAnswers, wrongAnswers, result } = quizSummary;
+  const [currentQuestion, setCurrentQuestion] = useState({});
+
+  const ShowCurrentQuestionDetails = ({currentQuestion}) => {
+    if (!Object.keys(currentQuestion).length) return false;
+    return (
+      <div className={`display-question ${currentQuestion.correct ? 'correct' : 'incorrect'}`}>
+        <div className="question-number">Question: {currentQuestion?.qNo}</div>
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/EndScreen.jsx:15]: until api is good currentQuestion cannot be undefinded

File: src/plays/fun-quiz/EndScreen.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            <h4>Wrong Answer(s): {wrongAnswers}</h4>
+          </div>
+          <div className='front-footer'>
+            <button onClick={redirectHome}>Home</button>
+          </div>
+          <div className='circle-area'>
+            {result.map((item, index) => {
+              return (
+                <div
+                key={item.question+index}
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/EndScreen.jsx]: done

File: src/plays/fun-quiz/EndScreen.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                >
+                  {index + 1}
+                </div>
+              );
+            })}
+          </div>
+          <ShowCurrentQuestionDetails currentQuestion={currentQuestion} />
+        </div>
+      </div>
+      <div></div>
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/EndScreen.jsx]: removed

File: src/plays/fun-quiz/FrontScreen.jsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  {
+    name: "Books",
+    id: 10,
+  },
+  {
+    name: "General Knwoledge",
+    id: 9,
+  },
+  {
+    name: "Film",
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/FrontScreen.jsx]: done

File: src/plays/fun-quiz/Readme.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# Fun Quiz
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/Readme.md]: added

File: src/plays/fun-quiz/QuizScreen.scss [hunk 7]
```diff
@@ -0,0 +1,222 @@
+@import url("https://fonts.googleapis.com/css2?family=Russo+One&display=swap"); // russo one
+
+$text-color: rgba(32, 73, 105, 1);
+
+.fun-quiz {
+  .fun-quiz-screen {
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/QuizScreen.scss]: done

File: src/plays/fun-quiz/QuizScreen.scss [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+    margin-top: 4rem;
+    width: 100%;
+    overflow: hidden;
+    position: relative;
+    color: black;
+    display: block;
+    font-family: "Quicksand", sans-serif;
+    position: relative;
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/QuizScreen.scss:18]: removed

File: src/plays/fun-quiz/QuizScreen.scss [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+    .loading-overlay {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+
+    }
+
+    .timer {
+      border: 1px solid #444444;
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/QuizScreen.scss]: done

**Review by koustov — COMMENTED** [review]
some questions added. also please notify once you are done with all comments.

File: src/plays/fun-quiz/QuizScreen.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (koustov) [reply src/plays/fun-quiz/QuizScreen.scss]: This is a test comment. Pls ignore

**Comment by Angryman18** [comment]
i dont see any issue with hover. we only want to show pointer when its hovered. so eitherway it works the same.
and i believe only questions would be coming from the api so i didnt add additional validiation there.
and for nullable, i dont understand if you are talking about current question if so then current question cannot be null. its defined with a value and some certain conditions are applied for updating it's value  (when timer gets timed out or user clicks "confirm" button or at veryfirst clicking "Let start" button).

**Comment by atapas** [comment]
> i dont see any issue with hover. we only want to show pointer when its hovered. so eitherway it works the same. and i believe only questions would be coming from the api so i didnt add additional validiation there. and for nullable, i dont understand if you are talking about current question if so then current question cannot be null. its defined with a value and some certain conditions are applied for updating it's value (when timer gets timed out or user clicks "confirm" button or at veryfirst clicking "Let start" button).

Look like we are close to getting done with the comments. Can't wait to see this one crossing the border...

@Angryman18 @koustov

**Comment by koustov** [comment]
@Angryman18 is it ready to be reviewed one last time?

File: src/plays/fun-quiz/FrontScreen.jsx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+        </div>
+      </>
+    );
+  };
+
+  const CategorySelector = () => {
+    return (
+      <>
+        <p>
+          The Quiz app reqires to have a specefic category in order to start
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/FrontScreen.jsx]: fixed

File: src/plays/fun-quiz/QuizScreen.scss [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+          font-size: 1rem;
+          padding: 0.5rem 1.5rem;
+          background-color: rgba(234,227,227,1);
+          border-radius: 40px;
+          transition: 100ms ease-in-out;
+          text-align: left;
+          vertical-align: middle;
+
+          &:hover {
+            cursor: pointer;
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/QuizScreen.scss]: fixed.

File: src/plays/fun-quiz/QuizScreen.scss [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+    .timer {
+      border: 1px solid #444444;
+      border-radius: 50%;
+      width: 100px;
+      aspect-ratio: 1/1;
+      display: grid;
+      place-items: center;
+      font-family: "Russo One", sans-serif;
+      font-size: 3rem;
+      color: #444444;
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/QuizScreen.scss]: fixed

File: src/plays/fun-quiz/EndScreen.jsx [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+  const { correctAnswers, cheatedAnswers, wrongAnswers, result } = quizSummary;
+  const [currentQuestion, setCurrentQuestion] = useState({});
+
+  const ShowCurrentQuestionDetails = ({currentQuestion}) => {
+    if (!Object.keys(currentQuestion).length) return false;
+    return (
+      <div className={`display-question ${currentQuestion.correct ? 'correct' : 'incorrect'}`}>
+        <div className="question-number">Question: {currentQuestion?.qNo}</div>
+        <li
+          dangerouslySetInnerHTML={{
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/plays/fun-quiz/EndScreen.jsx:21]: examples:
Which of these is NOT a team available in the game Pok&eacute;mon Go?
&quot;Minecraft&quot; was released from beta in 2011 during a convention held in which city?

**Comment by Angryman18** [comment]
i think the required changes are made.

**Review by koustov — APPROVED** [review]
LGTM

**Pull request merged**